src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CollectionBackedStoreProvider.cs

```diff
@@ -0,0 +1,159 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal class CollectionBackedStoreProvider : IStorePal
+    {
+        private readonly X509Certificate2Collection _certs;
+
+        internal CollectionBackedStoreProvider(X509Certificate2 cert)
+        {
+            _certs = new X509Certificate2Collection(cert);
+        }
+
+        internal CollectionBackedStoreProvider(X509Certificate2Collection certs)
+        {
+            _certs = new X509Certificate2Collection(certs);
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public IEnumerable<X509Certificate2> Find(X509FindType findType, object findValue, bool validOnly)
+        {
+            return Array.Empty<X509Certificate2>();
+        }
+
+        public byte[] Export(X509ContentType contentType, string password)
+        {
+            switch (contentType)
+            {
+                case X509ContentType.Cert:
+                    return ExportX509Der();
+                case X509ContentType.Pfx:
+                    return ExportPfx(password);
+                default:
+                    throw new CryptographicException();
```

Comment:
Ish.  Values other than that can get into here; but this object is never exposed outside of the library plumbing.  I can switch this back to yet another NotImplementedException to be cleaned up in the #1993 cleanup pass.  (The Serialized* forms are probably going to be PlatformNotSupportedException, because they're CAPI structures, not standards).

PKCS#7 will probably get around to appearing in beta8/rc.